File: Data Structures & Algorithms/two-integer-sum/submission-0.java


class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> map = new HashMap<>();
        int[] ans = new int[2];

        for(int i=0; i<nums.length; i+=1){
            int second = target - nums[i];
            if(map.containsKey(second)){
                ans[0] = map.get(second);
                ans[1] = i;
                return ans;
            }
            else{
                map.put(nums[i],i);
            }

        }

        return ans;
        
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        left = 0
        right = len(numbers)-1
        sum_total = numbers[left] + numbers[right]
        ans = []

        while sum_total != target:
            if sum_total>target:
                right-=1
            else:
                left+=1
            sum_total = numbers[left] + numbers[right]
        
        ans.append(left+1)
        ans.append(right+1)

        return ans




        